cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/FlowUtils.java

```diff
@@ -68,8 +69,13 @@ public static long generateConsumerGroupId(Program program, String flowletId) {
    * Generates a queue consumer groupId for the given flowlet in the given program id.
    */
   public static long generateConsumerGroupId(Id.Program program, String flowletId) {
+    // Use 'developer' in place of a program's namespace for programs in the 'default' namespace
+    // to support backwards compatibility for queues and streams.
+    String namespace = program.getNamespaceId();
+    String backwardsCompatibleNamespace =
+      Constants.DEFAULT_NAMESPACE.equals(namespace) ? Constants.DEVELOPER_ACCOUNT : namespace;
     return Hashing.md5().newHasher()
-                  .putString(program.getNamespaceId())
+                  .putString(backwardsCompatibleNamespace)
```

Comment:
So if someone creates a "developer" namespace and deploys an app with a flow, then that app could be reading from a stream already created in the "default" namespace?